### PR TITLE
check-gh-automation bug resolution

### PR DESCRIPTION
**Pull request by Toma5OD**
#### Overview
- **Files Modified:**
  - `/check-gh-automation/main.go`
  - `hack/local-check-gh-automation.sh`
  - `main_test.go` (additional changes)
- **Goal:** Ensure proper permissions for `openshift-cherrypick-robot` in repositories using the cherrypick plugin.

#### Changes
1. **`main.go`**: 
   - Added constants for cherrypick plugin and robot.
   - New field for plugin configuration in `options` struct.
   - Plugin flag parsing in `gatherOptions`.
   - Initialization for plugin configuration in `main`.
   - Enhanced `checkRepos` to validate `openshift-cherrypick-robot`.

2. **`local-check-gh-automation.sh`**:
   - Reordered function definitions.
   - New directory setup for plugin data.
   - Extracted plugin configuration.
   - Updated `go run` command for plugin-related flags.

3. **`main_test.go`**:
   - Adjustments to align with changes in `main.go`.

#### Impact
- Affects CI setups involving new or existing repositories with the cherrypick plugin.

#### Acceptance Criteria
- `openshift-cherrypick-robot` verified as an org member if cherrypick plugin is configured.

### CODE REVIEW

File: cmd/check-gh-automation/main.go [hunk 1]
```diff
@@ -190,6 +190,11 @@ func checkRepos(repos []string, bots []string, ignore sets.Set[string], client a
 
 		if configAgent != nil {
 			externalPlugins := configAgent.Config().ExternalPlugins[orgRepo]
+			repoLogger.Infof("is there a config for %s? %v", orgRepo, externalPlugins != nil)
```
Comment (smg247) [inline cmd/check-gh-automation/main.go:193]: remove this logging, this was only for testing

File: cmd/check-gh-automation/main.go [hunk 2]
```diff
@@ -190,6 +190,11 @@ func checkRepos(repos []string, bots []string, ignore sets.Set[string], client a
...
+			if externalPlugins == nil {
+				externalPlugins = configAgent.Config().ExternalPlugins[org]
+				repoLogger.Infof("is there a config for %s? %v", org, externalPlugins != nil)
```
Comment (smg247) [inline cmd/check-gh-automation/main.go:196]: remove this logging, this was only for testing

File: cmd/check-gh-automation/main.go [hunk 3]
```diff
@@ -205,6 +210,16 @@ func checkRepos(repos []string, bots []string, ignore sets.Set[string], client a
...
+		isCollaborator, err := client.IsCollaborator(org, repo, cherrypickRobot)
+		if err != nil {
+			return nil, fmt.Errorf("error checking collaborator status for openshift-cherrypick-robot on %s/%s: %w", org, repo, err)
+		}
+		if !isCollaborator {
+			repoLogger.Infof("openshift-cherrypick-robot is not a collaborator on %s/%s", org, repo)
+			failing.Insert(orgRepo)
+		} else {
+			repoLogger.Infof("Confirmed: openshift-cherrypick-robot is a collaborator on %s/%s", org, repo)
+		}
```
Comment (smg247) [inline cmd/check-gh-automation/main.go:222]: This section needs to take place inside the
```
if plugin.Name == cherrypickPlugin {
```

Comment (smg247) [reply cmd/check-gh-automation/main.go:222]: Also, keep in mind that the bot needs to either be a collaborator **or** an org member. It is not necessary for it to be both.

File: cmd/check-gh-automation/main.go (hunk 1 above)
Comment (Toma5OD) [reply cmd/check-gh-automation/main.go]: Removed

File: cmd/check-gh-automation/main.go (hunk 2 above)
Comment (Toma5OD) [reply cmd/check-gh-automation/main.go]: Removed

File: cmd/check-gh-automation/main.go (hunk 3 above)
Comment (Toma5OD) [reply cmd/check-gh-automation/main.go]: Restructured the code + add or check

**Pull request merged**